src/Columns/ColumnArray.h

```diff
@@ -213,6 +214,17 @@ class ColumnArray final : public COWHelper<IColumnHelper<ColumnArray>, ColumnArr
     ColumnPtr filterNullable(const Filter & filt, ssize_t result_size_hint) const;
     ColumnPtr filterGeneric(const Filter & filt, ssize_t result_size_hint) const;
 
+    /// Specializations for insertManyFrom
```

Comment:
Do you have any query example that really benefits from such specializations? IMHO from the code perspective such specializations looks like overkill and make sense only when there is a noticable improvement in some queries.

Reply:
![perf8](https://github.com/ClickHouse/ClickHouse/assets/8181003/e6059f59-4802-43e3-8452-54f1df473367)

Yes, it makes sense. This flamegraph was from our production environment. It is obvious that `insertManyFrom` takes most time in `MergeJoin::allInnerJoin`, which implements in partial merge join algorithm. Current implementation of `insertManyFrom` is too slow because the overhead of invoking `insertFrom` could not be ignored.

Reply:
Opts in ColumnArray introduce too much function calls, which make it very slow. I had removed those codes.